quickwit-index-config/src/default_index_config/default_config.rs

```diff
@@ -87,26 +103,50 @@ impl DefaultIndexConfigBuilder {
             let timestamp_field_entry = schema.get_field_entry(timestamp_field);
             if !timestamp_field_entry.is_fast() {
                 bail!(
-                    "Timestamp field must be a fast field, please add fast property to your field \
-                     `{}`.",
+                    "Timestamp field must be a fast field, please add the fast property to your \
+                     field `{}`.",
                     timestamp_field_name
                 )
             }
-            if let FieldType::I64(options) = timestamp_field_entry.field_type() {
-                if options.get_fastfield_cardinality() == Some(Cardinality::MultiValues) {
+            match timestamp_field_entry.field_type() {
+                FieldType::I64(options) | FieldType::U64(options) => {
+                    if options.get_fastfield_cardinality() == Some(Cardinality::MultiValues) {
+                        bail!(
+                            "Timestamp field cannot be an array, please change your field `{}` \
+                             from an array to a single value.",
+                            timestamp_field_name
+                        )
+                    }
+                }
+                _ => {
                     bail!(
-                        "Timestamp field cannot be an array, please change your field `{}` from \
-                         an array to a single value.",
+                        "Timestamp field must be either of type i64 or u64, please change your \
+                         field type `{}` to i64 or u64.",
                         timestamp_field_name
                     )
                 }
-            } else {
+            }
+        }
+
+        // Resolve sort field
+        let mut sort_by = None;
+        if let Some(sort_by_config) = self.sort_by {
```

Comment:
can we remove the `mut` ?

Either by using map, or by extracting the conversion in a function.

let sort_by = resolve_sort_field(...);